### PR TITLE
Add "Add to folders" action for bookmarks

### LINKED ISSUE #1228: Add "Copy to" action for bookmarks
**Is your feature request related to a problem? Please describe.**
This is related to #1171

**Describe the solution you'd like**

I want to be able to share my bookmarks with different teams while still having it in my "all bookmarks' view. To achieve this the right click menu should have a "copy to..." action which will allow the user to select which folder they want to copy it to. 

This allows the user to share his bookmark with multiple teams (each folder belonging to a different team) while still having it in his view.

**Describe alternatives you've considered**
See #1171

### LINKED ISSUE #1464: Show folder in bookmark details
**Is your feature request related to a problem? Please describe.**
When looking up bookmarks via tags, the bookmark is found.  However, it is just shown in a list.
There is no information available in which folder, or folders the bookmark is to be found.

**Describe the solution you'd like**
Show the related folders in the bookmark detail section.

**Describe alternatives you've considered**
I'm not aware of alternatives.

**Additional context**
NA (I think).

### PR DESCRIPTION
**Pull request by Artem4590**
**Now we can copy bookmarks with "Add to folders" action**

Exemple :

copying-bookmarks

Closes: #1228
Closes: #1464

Signed-off-by: Artem Lavrukhin <lavryha4590@gmail.com>

### CODE REVIEW

File: src/store/actions.js [hunk 1]
```diff
@@ -774,6 +803,54 @@ export default {
...
+		commit(mutations.FETCH_START, { type: 'copySelection' })
+		try {
+			await Parallel.each(
+				state.selection.folders,
+				async folder => {
+					if (folderId === folder.id) {
+						throw new Error('Cannot copy folder into itself')
+					}
+					const oldParent = folder.parent_folder
+					folder.parent_folder = folderId
```
Comment (marcelklehr) [inline src/store/actions.js:816]: This does not copy the folder. This should either be a real deep copy or the copy action shouldn't be available when folders are selected.

File: src/components/CopyDialog.vue [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			return this.$store.state.displayCopyDialog
+		},
+		selection() {
+			return this.$store.state.selection
+		},
+		title() {
+			if (this.selection.folders.length) {
+				if (this.selection.bookmarks.length) {
+					return n('bookmarks',
+						'Copyng %n folder and some bookmarks',
```
Comment (marcelklehr) [inline src/components/CopyDialog.vue:36]: Typo: "Copying"

**Review by marcelklehr — CHANGES_REQUESTED** [review]
Thank you for this contribution! The code is very well written and blends in with the existing code smoothly. :+1:

From a UX perspective, however, I'm not sold on calling this action "copy", as the bookmarks identity will stay the same and editing the "copied" bookmarks will change the original bookmarks as well.

**Comment by marcelklehr** [comment]
After a quick discussion with the design team, I'd preferably call this "Add to folders"

File: src/components/BulkEditing.vue [hunk 3]
```diff
@@ -16,7 +16,7 @@
 				</template>
 				{{ t('bookmarks', 'Move selection') }}
 			</ActionButton>
-			<ActionButton close-after-click @click="onBulkCopy">
+			<ActionButton v-if="!selectedFolders.length" close-after-click @click="onBulkCopy">
```
Comment (marcelklehr) [inline src/components/BulkEditing.vue:19]: :+1:

**Review by marcelklehr — APPROVED** [review]

**Comment by Artem4590** [comment]
Do I need to rename everything else to "Add to folders" ?

**Comment by marcelklehr** [comment]
> Do I need to rename everything else to "Add to folders" ?

I think it's fine if we call it copy in the code, as long as the UI is clear.

**Comment by Artem4590** [comment]
Sorry, the order in my name was not correct. Fix in this commit ☝🏻

**Pull request merged**